### ISSUE TITLE
Blank viewer

### ISSUE THREAD

**Issue by ppjiong11** [nerfstudio-project/nerfstudio#1735]
When I run nerfstudio on linux and open the viewer, I don't see the input image, just a blank coordinate system,please tell me why.
I can see training progress in the terminal。

**Comment by tancik** [comment]
Are you running locally or on a server? If on a server, are you forwarding the port? If locally, do you have a firewall? Can you send a screenshot of the viewer window and one of your terminal.

**Comment by ppjiong11** [comment]
I run nerfstudio on ubuntu locally, there is my screenshot
![Uploading 51C0EB5CD28F96F81107CAA768B43A85.jpg…]()
![A192490CF4ADA6C66F54B2699C43596A](https://user-images.githubusercontent.com/104246718/232184430-574598b1-d6a4-42e3-8306-047a4b456d12.jpg)

**Comment by ppjiong11** [comment]
![51C0EB5CD28F96F81107CAA768B43A85](https://user-images.githubusercontent.com/104246718/232185480-420285e5-8e4a-45bd-814a-2122315a7f5d.jpg)

**Comment by fanhongxing** [comment]
I encountered the same issue. It can be displayed normally when using version 0.1.19, but it shows blank after upgrading to version 0.2.0.

**Comment by tancik** [comment]
@fanhongxing What URL are you navigating to?

**Comment by fanhongxing** [comment]
> @fanhongxing What URL are you navigating to?

Thank you for your prompt reply.
0.2.0  https://viewer.nerf.studio/versions/23-04-10-0/?websocket_url=ws://localhost:7007
0.1.19  https://viewer.nerf.studio/versions/23-03-9-0/?websocket_url=ws://localhost:7007

**Comment by tancik** [comment]
@fanhongxing Are you using chrome? If so, can you go to `More Tools` -> `Developer Tools` -> `Console` and see if that reports any errors?

**Comment by tancik** [comment]
Can you also provide a screen shot.
Also what browser are you using?

**Comment by Zunhammer** [comment]
I'm having the same issue. Output of chrome dev tools console is:
![image](https://user-images.githubusercontent.com/18286310/232606379-35a9dfd7-9699-4637-afe5-1184fb850cc4.png)
on link: https://viewer.nerf.studio/versions/23-04-10-0/?websocket_url=ws://localhost:7007

On 0.1.19 it is working fine.

Besides, I noticed that in the install of ns-install-cli it says the follwoing:
```
bash: ../home/user/nerfstudio/nerfstudio/viewer/app/src/modules/WebSocket/ViserMessages.tsx: Permission denied
bash: Wrote: command not found
```
Looks at some point it is trying to move a folder up for a global path (../home/....)?

Thanks for your support!

**Comment by brentyi** [comment]
FYI the `ns-install-cli` error should be fixed by #1754. This should only impact the completions though.

As a datapoint I also just tried both 0.1.19 and 0.1.20 on both Chrome and Firefox; both are working on my end.

**Comment by tancik** [comment]
It worked for me when I just tried with a fresh install, ubuntu server, osx client, firefox and chrome.

**Comment by fanhongxing** [comment]
I am running on the server and have forwarded the port.
![image](https://user-images.githubusercontent.com/41686308/232640470-fcc6f129-65ae-4f88-bf76-b5d2d48827e8.png)
![image](https://user-images.githubusercontent.com/41686308/232640520-a9c093b5-9ea2-4a4c-a9b9-64f4a93ea762.png)
![image](https://user-images.githubusercontent.com/41686308/232640631-5d145265-6296-4a24-90cc-a98fab0ced9e.png)


> @fanhongxing Are you using chrome? If so, can you go to `More Tools` -> `Developer Tools` -> `Console` and see if that reports any errors?

**Comment by tancik** [comment]
@fanhongxing You are forwarding port `7007` to `8591`, correct?
How did you install the package, `pip install nerfstudio` or `pip install -e .`? If the later, can you try rerunning it.

**Comment by tancik** [comment]
Can you try modifying the following line from `host="localhost"` to `host="0.0.0.0"` to see if that fixes it?
https://github.com/nerfstudio-project/nerfstudio/blob/c1c7a3b2831defd0f3ffd4d6cb410f2454d124c5/nerfstudio/viewer/server/viewer_state.py#L125

**Comment by aiddun** [comment]
I'm also having the same issue right now, and that line change fixed it for me.

**Comment by Zunhammer** [comment]
Also fixes the problem for me :+1:

**Comment by fanhongxing** [comment]
> Can you try modifying the following line from `host="localhost"` to `host="0.0.0.0"` to see if that fixes it?
> 
> https://github.com/nerfstudio-project/nerfstudio/blob/c1c7a3b2831defd0f3ffd4d6cb410f2454d124c5/nerfstudio/viewer/server/viewer_state.py#L125

@tancik Thanks! Changing host from 'localhost' to '0.0.0.0' can solve the problem! Another possible solution is to set the address of the remote server in local port forwarding to '127.0.0.1', instead of the address of the SSH server.
![image](https://user-images.githubusercontent.com/41686308/232772990-a479432c-2a4e-4895-9d25-b57483deb7ad.png)

**Comment by Zunhammer** [comment]
In v0.2.1 the localhost was replaced by 127.0.0.1 instead of 0.0.0.0 which again gives me a blank viewer (disconnected, basically the same as before). The fix again is to replace 127.0.0.1 by 0.0.0.0 in line 125 in nerfstudio/viewer/server/viewer_state.py

**Comment by ppjiong11** [comment]
> Are you running locally or on a server? If on a server, are you forwarding the port? If locally, do you have a firewall? Can you send a screenshot of the viewer window and one of your terminal.

Does Ubuntu's firewall affect the viewer？could you please tell me ？

**Comment by tancik** [comment]
@ppjiong11 only if you are blocking the port (7007 by default). Try using the latest github version to see if it works.

**Comment by iraj465** [comment]
Hey i'm running the code in a remote server
<img width="941" alt="Screenshot 2023-04-24 at 3 48 44 AM" src="https://user-images.githubusercontent.com/32464366/233869170-55f3b81c-ba59-45b9-9778-6a917baf93f3.png">

I port forwarded port 7008 of remote with 7008 of my local: `ssh -L 7008:localhost:7008 user@remote_ip`
Getting this after i SSH;
<img width="584" alt="Screenshot 2023-04-24 at 3 50 50 AM" src="https://user-images.githubusercontent.com/32464366/233869234-884063f5-4d0b-4790-8d18-b1f9164a2c85.png">


Also simultaneously, the renderer looks like this.

<img width="1616" alt="Screenshot 2023-04-24 at 3 48 26 AM" src="https://user-images.githubusercontent.com/32464366/233869249-6d8e3525-5218-49d5-a3c0-b0b0bc871f77.png">

How to resolve this? Can't seem to understand what's breaking

**Comment by tancik** [comment]
@iraj465 It looks like the port is being blocked. This could be a firewall on the network port, or you restrictions on your network. Are you using a corporate network by chance?

**Comment by iraj465** [comment]
@tancik Yes it's a corporate network, but i'm usually able to port forward and run other apps from remote server to local.
I have the viewer app running in remote server.
<img width="528" alt="Screenshot 2023-04-26 at 4 09 41 AM" src="https://user-images.githubusercontent.com/32464366/234420118-3b41e27d-3eff-4242-9708-6d68ad72fd0a.png">
<img width="1031" alt="Screenshot 2023-04-26 at 4 09 28 AM" src="https://user-images.githubusercontent.com/32464366/234420304-7552803a-cee5-4391-ba6b-78b680b826df.png">


After port forwarding, i'm getting this error.
<img width="1621" alt="Screenshot 2023-04-26 at 4 09 12 AM" src="https://user-images.githubusercontent.com/32464366/234420587-5f3822c0-2c9a-430f-a815-f077e6496d14.png">

**Comment by brentyi** [comment]
@iraj465 have you tried the latest / GitHub version of nerfstudio, or are you use the PyPI version?

The latest releases are still missing #1779, which may fix some networking issues.

**Comment by iraj465** [comment]
@brentyi  i'm using the current Github version and installed it with `pip install -e .`
I also changed the host address to `0.0.0.0` as suggested but still it seems to not render in my local after port forwarding.

**Comment by tancik** [comment]
@iraj465 Are you using chrome? There are some issues connecting with websockets when using brave or safari.

**Comment by flow-specter** [comment]
Hi guys, how to change the port from 7007 to another one, like 7008?

**Comment by brentyi** [comment]
You can pass in ` --viewer.websocket-port`, eg:
```
ns-train nerfacto  --viewer.websocket-port 7008
```
